pennylane/math/utils.py

```diff
@@ -570,3 +579,35 @@ def in_backprop(tensor, interface=None):
         return False
 
     raise ValueError(f"Cannot determine if {tensor} is in backpropagation.")
+
+
+def is_non_scalar_tensor(arg) -> bool:

```

Comment:
What was the motivation behind putting this in `qml.math`?

Reply:
I initially put that into the `capture_qnode.py` file, but then I realized that `qml.math` might have been more appropriate (I can also put it back though if it is an issue)

Reply:
I'm pretty agnostic to it. `qml.math` might make sense since this seems like a utility function that can be useful in other modules as well. Could you also add some unit tests for it?

Reply:
I kinda feel like it would be a better private helper at the moment.  I'm not sure it's particularly needed generally, and I don't think we're hiding any complexity here.